### PR TITLE
feat: add documentation workflow

### PR DESCRIPTION
**Pull request by skjnldsv**
See https://github.com/nextcloud/nextcloud-event-bus/pull/606

Mostly used for typescript libraries

### CODE REVIEW

File: workflow-templates/documentation.yml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+    name: Build and deploy
+    steps:
+      - name: Check actor permission level
+        uses: skjnldsv/check-actor-permission@e591dbfe838300c007028e1219ca82cc26e8d7c5 # v2
+        with:
+          require: admin
+
+      - name: Checkout
+        uses: actions/checkout@ac593985615ec2ede58e132d2e21d2b1cbd6127c # v3
```
Comment (nickvergessen) [inline workflow-templates/documentation.yml:20]: update to 3.5.2 here similarly to #157

**Review by nickvergessen — APPROVED** [review]
Otherwise fine

File: workflow-templates/documentation.yml (hunk 1 above)
Comment (skjnldsv) [inline workflow-templates/documentation.yml:20]:
```suggestion
        uses: actions/checkout@8e5e7e5ab8b370d6c329ec480221332ada57f0ab # v3.5.2
```

File: workflow-templates/documentation.yml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+      - name: Read package.json node and npm engines version
+        uses: skjnldsv/read-package-engines-version-actions@0ce2ed60f6df073a62a77c0a4958dd0fc68e32e7 # v2.1
+        id: versions
+        with:
+          fallbackNode: '^16'
+          fallbackNpm: '^8'
+
+      - name: Set up node ${{ steps.versions.outputs.nodeVersion }}
+        uses: actions/setup-node@8c91899e586c5b171469028077307d293428b516 # v3
```
Comment (skjnldsv) [inline workflow-templates/documentation.yml:30]:
```suggestion
        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3.6.0
```

File: workflow-templates/documentation.yml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+      - name: Install dependencies & build
+        run: |
+          npm ci
+          npm run build --if-present
+          npm run build:doc --if-present
+
+      - name: Deploy
+        # Only deploy on release
+        if: github.event.release
+        uses: peaceiris/actions-gh-pages@373f7f263a76c20808c831209c920827a82a2847 # v3
```
Comment (skjnldsv) [inline workflow-templates/documentation.yml:46]:
```suggestion
        uses: peaceiris/actions-gh-pages@373f7f263a76c20808c831209c920827a82a2847 # v3.9.3
```

**Review by nickvergessen — APPROVED** [review]

**Pull request merged**